src/Data/Core.Caching/CacheStoreType.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Runtime.Serialization;
+
+namespace QuantumCore.Caching;
+
+public enum CacheStoreType
+{
+    
+    /// <summary>
+    /// Redis database for common data
+    /// </summary>
+    [EnumMember(Value = "Common")]
+    Shared = 0,
+    
+    /// <summary>
+    /// Redis database for current server
+    /// </summary>
+    [EnumMember(Value = "Server")]
+    Server = 1, // todo: read from config
```

Comment:
Read what from config?

Reply:
Sorry, I could have been more specific with the comment line. For considering the multiple servers, there will be a need for redis database seperation between the so told servers. When creating the redis client, If our `CacheStoreType` case is `Server`, we could go to config and read the database index out of 16. Doing so, it gives us an opportunity for managing 15 game servers in just one redis instance.

Reply:
I see. Where does the magic `16` come from? Why only 16 servers in one redis instance? Should we design the code to talk to 1 redis cluster and let redis handle the instancing?

Reply:
By default redis sets it to 16, but supports as many databases as we want, we can configure it in `redis.conf` with `databases` key.  In principle, Redis databases on the same instance are no different than collections in mongodb.

[redis.conf:381](https://github.com/redis/redis/blob/unstable/redis.conf#L381):
```
# Set the number of databases. The default database is DB 0, you can select
# a different one on a per-connection basis using SELECT <dbid> where
# dbid is a number between 0 and 'databases'-1
databases 16
```

> Should we design the code to talk to 1 redis cluster and let redis handle the instancing?
That's up to how we want to approach distributed servers. Do we really want to create a redis instance per server?

Reply:
I still don't get what redis databases are. Sorry until now I used redis only as distributed cache in ASP.NET Core - never redis itself.

> That's up to how we want to approach distributed servers. Do we really want to create a redis instance per server?

I though that we use any redis instance (for example a cloud instance) that scales automatically?

As I'm not a redis pro I leave this stuff to you. Do however you think is the best :)

Reply:
No worries, I've got a humble experience on redis and happy to share. So basically, redis is a memory based no-sql database. Restarting redis instance _(without [`BGSAVE` command](https://redis.io/docs/latest/commands/bgsave/) on)_ will wipe all the data stored in the instance. With this in mind, auto-scaling is not a healthy option anymore because every scaling triggers a wipe, we do not want that. 

I've worked with redis at large scales (~15-20m. key per db) and saw that redis handles 16 databases impressingly fast, I can't say "the more than merrier" in this case, bc idk the consequences, lol. Feel free to ask more :)

Reply:
I think I understand enough for this PR. Will come back to you and ask some redis questions :)

Feel free to take ownership of the redis/caching part 👌